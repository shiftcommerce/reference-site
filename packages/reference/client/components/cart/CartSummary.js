// Libraries
import { Component } from 'react'
import Link from 'next/link'

// Libs
import { calculateCartSummary } from '../../lib/calculateCartSummary'
import { fixedPrice } from '../../lib/fixedPrice'

// Objects
import Button from '../../objects/Button'

class CartSummary extends Component {
  shippingText () {
    if (typeof window === 'undefined' || window.location.pathname !== '/checkout') {
      // TODO: make this dynamic in future
      return <>
        <a>Estimated shipping cost: &pound;3.45</a>
      </>
    } else {
      return <>
         <a>Shipping cost:</a>
       </>
    }
  }

  render () {
    const totals = calculateCartSummary(this.props.cart)

    return (
      <>
      <section className='c-cart-summary'>
        <dl>
          <dt><a>Total Price:</a></dt>
          <dd><a>&pound;{fixedPrice(totals.subTotal)}</a></dd>
        </dl>
        <dl>
          <dt>{ this.shippingText() }</dt>
        </dl>
        <dl>
          <dt><a className='u-bold'>TOTAL:</a></dt>
          <dd><a className='u-bold'>&pound;{fixedPrice(totals.total)}</a></dd>
        </dl>
        <dl>
          <a>* Including VAT</a>
        </dl>
      </section>
      <section className='c-cart-summary__buttons'>
        <Button
          aria-label='Continue Shopping'
          label='Continue shopping'
          size='lrg'
          status='grey'
          className='c-cart-summary__buttons--continue'
          type='button'
        />
        <Link href='/checkout' as=''>
          <Button
            aria-label='Place Order'
            label='Continue to payment'
            size='lrg'
            className='c-cart-summary__buttons--proceed'
            status='positive'
            type='button'
          />
        </Link>
      </section>
      </>
    )
  }
}

export default CartSummary
