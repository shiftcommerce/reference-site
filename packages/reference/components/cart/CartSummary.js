// Libraries
import { Component } from 'react'
import { calculateCartSummary } from '../../lib/calculateCartSummary'

class CartSummary extends Component {
  render () {
    const { subTotal, total, tax, discount } = calculateCartSummary(this.props.cart)

    return (
      <section className='c-cart-summary'>
        <dl>
          <dt> Sub total: </dt>
          <dd> &pound;{ subTotal } </dd>
        </dl>
        <dl>
          <dt> Tax applied: </dt>
          <dd className='u-text-color--red'> +&pound;{ tax } </dd>
        </dl>
        <dl>
          <dt> Amount you saved on this purchase: </dt>
          <dd className='u-text-color--green'> -&pound;{ discount } </dd>
        </dl>
        <dl>
          <dt> Total amount to be paid: </dt>
          <dd> <b>&pound;{ total }</b> </dd>
        </dl>
      </section>
    )
  }
}

export default CartSummary
