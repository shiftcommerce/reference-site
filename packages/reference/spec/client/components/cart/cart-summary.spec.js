import CartSummary from '../../../../client/components/cart/cart-summary'

import cartFixture from '../../../fixtures/cart'

test('renders correct values when line items are available', () => {
  // arrange
  const cart = {
    lineItems: [],
    sub_total: 0,
    discount_summaries: [],
    shipping_total: 0
  }

  // act
  const wrapper = mount(
    <CartSummary cart={cart} />
  )

  // assert
  expect(wrapper).toMatchSnapshot()
  expect(wrapper).toIncludeText('Total Price:£0')
  expect(wrapper).toIncludeText('Estimating shipping cost...')
  expect(wrapper).toIncludeText('Total Price:£0.00')
})

test('renders correct values when line items and discounts are available', () => {
  // act
  const wrapper = mount(
    <CartSummary cart={cartFixture} />
  )

  // assert
  expect(wrapper).toMatchSnapshot()
  expect(wrapper).toIncludeText('Total Price:£6.77')
  expect(wrapper).toIncludeText('£1.50 off all orders')
  expect(wrapper).toIncludeText('TOTAL:£5.27')
})

test('displays placeholder text, fetches shipping methods and sets the estimated cost to the cheapest one', async () => {
  cartFixture.shipping_method = undefined
  const fetchShippingSpy = jest.spyOn(CartSummary, 'fetchShippingMethods').mockImplementation(() => Promise.resolve({
    data: [{
      id: 1,
      sku: 'STA_SHIP',
      label: 'Standard shipping',
      total: 3.75,
      meta_attributes: {
        working_days: {
          value: 2
        }
      }
    }, {
      id: 2,
      sku: 'NEX_SHIP',
      label: 'Next day delivery',
      total: 6.25,
      meta_attributes: {
        working_days: {
          value: 1
        }
      }
    }]
  }))

  const wrapper = mount(
    <CartSummary cart={cartFixture} />
  )

  expect(wrapper).toIncludeText('Estimating shipping cost...')

  await wrapper.instance().componentDidMount()
  wrapper.update()

  expect(wrapper).toIncludeText('Shipping from: £3.75')

  fetchShippingSpy.mockRestore()
})
