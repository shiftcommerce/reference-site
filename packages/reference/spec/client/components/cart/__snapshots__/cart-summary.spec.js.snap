// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correct values when line items are available 1`] = `
<CartSummary
  cart={
    Object {
      "lineItems": Array [
        Object {
          "discount": "2.00",
          "imageUrl": "",
          "price": "10.00",
          "quantity": 2,
          "size": "size - 8",
          "sku": "123",
          "title": "test",
        },
      ],
    }
  }
  updateQuantity={[Function]}
>
  <section
    className="c-cart-summary"
  >
    <dl>
      <dt>
        <a>
          Total Price:
        </a>
      </dt>
      <dd>
        <a>
          £
          20.00
        </a>
      </dd>
    </dl>
    <dl>
      <dt>
        <a>
          Estimated shipping cost: £3.45
        </a>
      </dt>
    </dl>
    <dl>
      <dt>
        <p
          className="u-bold"
        >
          TOTAL:
        </p>
      </dt>
      <dd>
        <p
          className="u-bold"
        >
          £
          19.45
        </p>
      </dd>
    </dl>
    <dl>
      <a>
        * Including VAT
      </a>
    </dl>
  </section>
  <section
    className="c-cart-summary__buttons"
  >
    <Link
      as="/"
      href="/slug?slug=/homepage"
    >
      <a
        className="o-button--sml c-cart-summary__buttons--continue"
        href="/"
        onClick={[Function]}
      >
        continue shopping
      </a>
    </Link>
    <Link
      href="/checkout"
    >
      <a
        className="o-button--sml c-cart-summary__buttons--proceed"
        href="/checkout"
        onClick={[Function]}
      >
        Checkout
      </a>
    </Link>
  </section>
</CartSummary>
`;

exports[`renders correct values when no line items are available 1`] = `
<CartSummary
  cart={
    Object {
      "lineItems": Array [],
    }
  }
>
  <section
    className="c-cart-summary"
  >
    <dl>
      <dt>
        <a>
          Total Price:
        </a>
      </dt>
      <dd>
        <a>
          £
          0.00
        </a>
      </dd>
    </dl>
    <dl>
      <dt>
        <a>
          Estimated shipping cost: £3.45
        </a>
      </dt>
    </dl>
    <dl>
      <dt>
        <p
          className="u-bold"
        >
          TOTAL:
        </p>
      </dt>
      <dd>
        <p
          className="u-bold"
        >
          £
          3.45
        </p>
      </dd>
    </dl>
    <dl>
      <a>
        * Including VAT
      </a>
    </dl>
  </section>
  <section
    className="c-cart-summary__buttons"
  >
    <Link
      as="/"
      href="/slug?slug=/homepage"
    >
      <a
        className="o-button--sml c-cart-summary__buttons--continue"
        href="/"
        onClick={[Function]}
      >
        continue shopping
      </a>
    </Link>
    <Link
      href="/checkout"
    >
      <a
        className="o-button--sml c-cart-summary__buttons--proceed"
        href="/checkout"
        onClick={[Function]}
      >
        Checkout
      </a>
    </Link>
  </section>
</CartSummary>
`;
